src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_baron_geddon.cpp

```diff
@@ -184,10 +203,51 @@ class spell_geddon_inferno : public SpellScriptLoader
     }
 };
 
+// 20478 Armageddon
+class spell_geddon_armageddon : public SpellScriptLoader
+{
+public:
+    spell_geddon_armageddon() : SpellScriptLoader("spell_geddon_armageddon") { }
+
+    class spell_geddon_armageddon_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_geddon_armageddon_AuraScript);
+
+        void HandleAfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (Creature* pCreatureTarget = GetTarget()->ToCreature())
+            {
+                pCreatureTarget->SetReactState(REACT_PASSIVE);
+                pCreatureTarget->AttackStop();
+            }
+        }
+
+        void HandleAfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (Creature* pCreatureTarget = GetTarget()->ToCreature())
+            {
+                pCreatureTarget->SetReactState(REACT_AGGRESSIVE);
```

Comment:
Asking without knowing, should Geddon restore the old target he was attacking?

Reply:
This hack is kinda ugly to be fair, but since we are working around the clock I suppose it's fine

Reply:
well, i wanted to add some correctness to aura application, i can add basic "hasAura" checks or similar timers but thats also... Other than that, Geddon will restore its target from threat manager here (if he still is in combat)